skiplist created with NewWithMaxLevel(19+) panics on Set

```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/sean-public/fast-skiplist.(*SkipList).getPrevElementNodes(0xc42006e120, 0x0, 0xc42006e178, 0xc420067500, 0x7f793c60f000)
        /home/user/go/src/github.com/sean-public/fast-skiplist/skiplist.go:109 +0xd9
github.com/sean-public/fast-skiplist.(*SkipList).Set(0xc42006e120, 0x0, 0x468de0, 0x4de9b0, 0x0)
        /home/user/go/src/github.com/sean-public/fast-skiplist/skiplist.go:28 +0x8a
main.main()
        /home/user/test.go:5 +0x59
exit status 2
```
The code (`test.go`):
```
package main
import "github.com/sean-public/fast-skiplist"
func main(){
  s := skiplist.NewWithMaxLevel(19)
  s.Set(0, struct{}{}) // line 5
}
```
`DefaultMaxLevel` (`18`) is used instead of `MaxLevel` in lines 160, 161, and 165:

https://github.com/sean-public/fast-skiplist/blob/1de4bd306c3f816ec0c2702c89d3e2f279f67422/skiplist.go#L154-L167

--- a/skiplist.go
+++ b/skiplist.go
@@ -157,12 +157,12 @@ func NewWithMaxLevel(maxLevel int) *SkipList {
 	}
 
 	return &SkipList{
-		elementNode:    elementNode{next: make([]*Element, DefaultMaxLevel)},
-		prevNodesCache: make([]*elementNode, DefaultMaxLevel),
+		elementNode:    elementNode{next: make([]*Element, maxLevel)},
+		prevNodesCache: make([]*elementNode, maxLevel),
 		maxLevel:       maxLevel,
 		randSource:     rand.New(rand.NewSource(time.Now().UnixNano())),
 		probability:    DefaultProbability,
-		probTable:      probabilityTable(DefaultProbability, DefaultMaxLevel),
+		probTable:      probabilityTable(DefaultProbability, maxLevel),
 	}
 }
 

--- a/skiplist_test.go
+++ b/skiplist_test.go
@@ -142,6 +142,11 @@ func TestChangeLevel(t *testing.T) {
 	}
 }
 
+func TestMaxLevel(t *testing.T) {
+	list := NewWithMaxLevel(DefaultMaxLevel + 1)
+	list.Set(0, struct{}{})
+}
+
 func TestChangeProbability(t *testing.T) {
 	list := New()
 

